src/hammer-tech/hammer_tech.py

```diff
@@ -811,7 +811,7 @@ def process_library_filter(self,
         # Next, extract paths and prepend them to get the real paths.
         def get_and_prepend_path(lib: Library) -> Tuple[Library, List[str]]:
             paths = filt.paths_func(lib)
-            full_paths = list(map(lambda path: self.prepend_dir_path(path, lib), paths))
+            full_paths = list(map(lambda path: self.prepend_dir_path(path, lib) if path else '', paths))
```

Comment:
Why did this change? Is this just a bugfix?

Reply:
prepend_dir_path has an assertion  that checks to make sure the path is not empty. I am not entirely sure why the assertion exists, but this was my solution to get an empty list to be returned

Reply:
Hmm, how do empty paths end up showing up to begin with? And how is this cap table specific?

Reply:
Well because you only need a cap_table_file OR a qrc_techfile then 1 of them can be empty. Previously hammer would error if you did not supply a qrc_techfile even though the function "generate_mmmc_script" already assumed that if there was no supplied qrc_techfile that an empty string would be retrieved when calling "get_mmmc_wrc". I guess this raises the question of whether we actually want an error to be thrown if neither of these files are supplied

Reply:
Hm, seeing as how other things use `prepend_dir_path` I also don't see what the value of the `len(path) > 0` assertion is, seeing as it is only used in certain cases (DRC/LVS decks, map files) but doesn't make whatever you specified in the tech json any more or less correct.

Seems like what we really should do is call an `os.path.exists` on the constructed path so that Python can catch bad paths in tech jsons before it's passed to the tool where sometimes at best only a warning is thrown.

Reply:
If there are no qrc techfiles, then wouldn't `get_mmmc_qrc()` return an empty string since the `read_libs` call would return an empty list?

Reply:
As a side note, we really need to detach the QRC techfile from the libraries (#315).

Reply:
Maybe we can incorporate that larger change into this PR, so that we aren't creating more work we need to undo/fix later?